### ISSUE TITLE
[BUG] Error - connect_transport_exception handshake failed because connection reset

### ISSUE THREAD

**Issue by bjo004** [opensearch-project/cross-cluster-replication#359]
**Describe the bug**
In creating two opensearch clusters (leader is called opensearch and follower is called opensearch-rep) via helm on Azure Kubernetes Service, we generated our own certificates, enabled security plugin and also the cross cluster replication plugin. We followed the guide on https://opensearch.org/docs/latest/replication-plugin/get-started/ we always get stuck.

##### ADDED LEADER ALIAS 
curl -XPUT -k -H 'Content-Type: application/json' -u 'USERNAME:PASSWORD' 'https://opensearch-rep.northeurope.cloudapp.azure.com:9200/_cluster/settings?pretty' -d '
{
  "persistent": {
    "cluster": {
      "remote": {
        "opensearch-leader-alias": {
          "seeds": ["opensearch.northeurope.cloudapp.azure.com:9300"]
        }
      }
    }
  }
}'

##### CREATE INDEX ON LEADER
curl -XPUT -k -H 'Content-Type: application/json' -u 'USERNAME:PASSWORD' 'https://opensearch.northeurope.cloudapp.azure.com:9200/opensearch-leader-index-01?pretty'

##### START REPLICATION
curl -XPUT -k -H 'Content-Type: application/json' -u 'USERNAME:PASSWORD' 'https://opensearch-rep.northeurope.cloudapp.azure.com:9200/_plugins/_replication/opensearch-rep-01/_start?pretty' -d '
{
   "leader_alias": "opensearch-leader-alias",
   "leader_index": "opensearch-leader-index-01",
   "use_roles":{
      "leader_cluster_role": "all_access",
      "follower_cluster_role": "all_access"
   }
}'

##### ERROR RECIEVED #####
{
  "error" : {
    "root_cause" : [
      {
        "type" : "transport_exception",
        "reason" : "handshake failed because connection reset"
      }
    ],
    "type" : "connect_transport_exception",
    "reason" : "[][52.142.87.134:9300] general node connection failure",
    "caused_by" : {
      "type" : "transport_exception",
      "reason" : "handshake failed because connection reset"
    }
  },
  "status" : 500
}
#######################

**Expected behavior**
Replication should start. 

**Plugins**
opensearch-alerting                 
opensearch-anomaly-detection        
opensearch-asynchronous-search      
opensearch-cross-cluster-replication
opensearch-index-management         
opensearch-job-scheduler            
opensearch-knn                      
opensearch-ml                       
opensearch-observability            
opensearch-performance-analyzer     
opensearch-reports-scheduler        
opensearch-security                 
opensearch-sql                      

**Host/Environment (please complete the following information):**
 - OS: [Kubernetes 1.21.9 & Helm 3.8.0]
 - OpenSearch Version [1.3.1]

**Additional context**
If it's a certificate issue, I also followed the guide https://opensearch.org/docs/latest/security-plugin/configuration/generate-certificates/

Any thoughts anyone? 
@rohin 
@saikaranam-amazon 

Kind regards,

Bankole.

**Comment by saikaranam-amazon** [comment]
@bjo004 Could you please attach the logs at the follower cluster to under the error?

Based on the error response, this is generic exception thrown at the client(remote cluster) and not replication plugin specific issue.

```
"type" : "transport_exception",
"reason" : "handshake failed because connection reset"
}
},
"status" : 500
```

Other steps to diagnose further:
- Can you invoke direct search request to the leader index from the follower cluster to identify any setup level issues?
 ex: `GET /opensearch-leader-alias:opensearch-leader-index-01/_search?pretty`

**Comment by bjo004** [comment]
Hi @saikaranam-amazon , thanks for your response. I tore down the cluster. I can bring it back up. Are you free for a quick session? Teams, Skype, Zoom? 

Kind regards,

Bankole.

**Comment by bjo004** [comment]
Hi @saikaranam-amazon I managed to get the cross cluster replication working with security enabled by doing the following. 
1) Generating certs for the leader cluster
2) Copying those certs to the follower cluster
3) Making the nodes_dn field the same on both leader and follower clusters

That fixed the transport_exception error and it's now replicating. 

Thanks for your response. 

I've got a new issue though. I'm getting timeouts. I'll open a new issue.

Kind regards,

Bankole.
